async-openai/src/config.rs

```diff
@@ -104,7 +95,7 @@ impl Config for OpenAIConfig {
     }
 
     fn api_key(&self) -> &str {
-        &self.api_key
+        &self.api_key.expose_secret()
```

Comment:
This actually leaks the protection proved by `Secret<String>` in first place. 

This calls for a change to function signature to  `fn api_key() -> &Secret<String>` in `Config` trait and in both implementations.

Reply:
Ah, ofc, https://github.com/64bit/async-openai/pull/114/commits/f5dade574e42d7770f9355081aa78023873c2849